src/main/scala/calc/Compiler.scala

```diff
@@ -74,6 +80,21 @@ object Compiler {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
+      case BinaryOp(op, rhs, lhs) =>
+        irt.BinaryOp(binaryOpMap(op), compileExpr(rhs), compileExpr(lhs))
+
+      case Let(name, value, body) =>
+        identMap.put(name, value)
```

Comment:
There are two problems with this approach:
* `value` will be evaluated every time `name` is used in `body`, which is not really the spirit of a `let`: `value` should be evaluated once now and store in a local variable, then that variable should be reused in the `body`.
* The global mutable map will not support a) concurrent calls to `compileExpr` nor b) nested `let`s that happen to use the same `name` (where inner `let`s should shadow outer `let`s only within their body).